src/plots/cartesian/axes.js

```diff
@@ -2345,12 +2344,20 @@ axes.drawOne = function(gd, ax, opts, allDepths) {
     // to determine how much to shift this one out by
     // TODO: Also need to account for the expected depth of the current axis
     // (if drawing from the left inwards)
-    if(axLetter === 'y') {
+    if(axLetter === 'y' & !ax.position > 0) {
```

Comment:
I am not sure if understand the statement here.
`!ax.position` would be `true` or `false` and we are comparing it to be greater than zero? 
Let's rewrite this statement.